filebeat/harvester/util.go

```diff
@@ -4,18 +4,20 @@ import "github.com/elastic/beats/libbeat/common/match"
 
 // Contains available prospector types
 const (
-	LogType   = "log"
-	StdinType = "stdin"
-	RedisType = "redis"
-	UdpType   = "udp"
+	LogType    = "log"
+	StdinType  = "stdin"
+	RedisType  = "redis"
+	UdpType    = "udp"
+	DockerType = "docker"
 )
 
 // ValidType of valid input types
 var ValidType = map[string]struct{}{
-	StdinType: {},
-	LogType:   {},
-	RedisType: {},
-	UdpType:   {},
+	StdinType:  {},
```

Comment:
@kvch Not related to this PR, but as we register the prospectors now I'm curious if we still need this?

Reply:
`ValidType` is still used in `log` prospector for config validation and opening the right input.
But there is no other check/use of it. The check now happens when a factory is retrieved from registry using `GetFactory`. So it is not required, but this part needs further refactoring to make `ValidType` `log` specific. For the sake of consistency and to make refactoring easier, it is acceptable to add docker to the list.

Reply:
I pushed https://github.com/elastic/beats/pull/5537 to remove this part.